### PR TITLE
setup sphinx doc

### PR DESCRIPTION
**Pull request by RNAer**
Address #31. @mortonjt , could you review it? @ElDeveloper , you have much more expertise on sphinx. It would be greate if you can take a quick look as well if it is not too much trouble for you.

note that the docstring follows the numpydoc convention, but it does not require numpydoc to generate the doc because the sphinx v1.3 has napolean extension to parse numpy-style or google style docstring.

And finally, I wanna to say it loudly - I hate sphinx...

### CODE REVIEW

**Comment by mortonjt** [comment]
Would we want to add the doc building to travis?

**Comment by RNAer** [comment]
Yes, I will. have you played with this PR? how do you like the generated
documentation?

A note:

because this issue of sphinx v1.3 (
https://github.com/sphinx-doc/sphinx/pull/1828), the doc build requires
sphinx 1.3.3 (not tested for v1.3.2). Interestingly, the latest in pypi is
v1.3.1. So you need to install from github releases:

`pip install https://github.com/sphinx-doc/sphinx/archive/1.3.3.tar.gz`

**Comment by mortonjt** [comment]
I like the rendering!

Noticed that some the default parameters don't render very well (i.e. TmpDir='/var/folders/7x/j817pb0j0pzdsgpk5rp3wp_w0000gp/T' in Prodigal)

**Comment by RNAer** [comment]
I noticed that too. that's from burrito, unfortunately. I think burrito
needs an overhaul, there are other minor, if not major, issues with burrito.

File: doc/Makefile [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# Makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
```
Comment (ElDeveloper) [inline doc/Makefile]: We've found it very useful to bump sphinx warnings to errors, so we add the `-W` option here. This usually prevent silent mistakes from going into master and then cascading into major annoyances later on.

File: micronota/bfillings/__init__.py [hunk 2]
```diff
@@ -1,3 +1,35 @@
+r'''
+Application Wrappers
+====================
+
+.. currentmodule:: micronota.bfillings
+
+This module (:mod:`micronota.bfillings`) provides wrappers of `class`
+for external bioinformatic tools such as `Prodigal`, 'HMMer`, etc.
```
Comment (ElDeveloper) [inline micronota/bfillings/__init__.py]: Can you replace the single quote here for a back tick?

**Comment by ElDeveloper** [comment]
Just those two comments.

**Comment by RNAer** [comment]
Good catches! will fix them. let me know what you think after you render
it. Thanks!

**Comment by ElDeveloper** [comment]
Rendering looks good to me.

**Pull request merged**